### PR TITLE
[MRG] FAST_COMPILE now lazy by default

### PR DESCRIPTION
**Pull request by nouiz**
Also make DebugMode not check strides by default.

### CODE REVIEW

File: theano/gof/vm.py [hunk 1]
```diff
@@ -705,6 +705,8 @@ def __init__(self, allow_gc=None, use_cloop=False, callback=None,
             version. If lazy is True or False, we force the version used
             between Loop/LoopGC and Stack.
 
+        c_thunks - If None or True, don't change the default. If False,
+            don't compile c code for the thunks.
         """
         # Note: if more parameters are added to __init__, make sure to forward
         # them in the "type(self)(...)" call in the "accept" method below.
```
Comment (lamblin) [inline theano/gof/vm.py]: As this comment says, you have to add "c_thunks" to the list of keywords passed to `type(self)(...)` at line 754.

**Comment by lamblin** [comment]
This line https://github.com/nouiz/Theano/blob/default_mode/theano/gof/vm.py#L107 should probably be:

    if reuse_out is not None:

**Comment by nouiz** [comment]
I updated the PR. I think we should way for the python2/3 PR to get merged before merging this one.

**Comment by nouiz** [comment]
I updated this PR to fix error in travis

**Comment by lamblin** [comment]
I'm OK to merge after python3 support is merged.

**Comment by lamblin** [comment]
Triggering a new merge just to be sure.

**Pull request merged**